zingo-testutils/src/lib.rs

```diff
@@ -737,6 +738,131 @@ pub mod scenarios {
+        // received from a faucet to sapling
+        faucet
+            .do_send(vec![(
+                &get_base_address!(recipient, "sapling"),
+                value.checked_div(4).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // received from a faucet to transparent
+        faucet
+            .do_send(vec![(
+                &get_base_address!(recipient, "transparent"),
+                value.checked_div(4).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // send to a faucet
+        recipient
+            .do_send(vec![(
+                &get_base_address!(faucet, "unified"),
+                value.checked_div(10).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // send to self orchard
+        recipient
+            .do_send(vec![(
+                &get_base_address!(recipient, "unified"),
+                value.checked_div(10).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // send to self sapling
+        recipient
+            .do_send(vec![(
+                &get_base_address!(recipient, "sapling"),
+                value.checked_div(10).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // send to self transparent
+        recipient
+            .do_send(vec![(
+                &get_base_address!(recipient, "transparent"),
+                value.checked_div(10).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // shield transparent
+        recipient
+            .do_shield(&[Pool::Transparent], None)
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // upgrade sapling
+        recipient.do_shield(&[Pool::Sapling], None).await.unwrap();
+        // end
+        scenario_builder
+            .regtest_manager
+            .generate_n_blocks(4)
```

Comment:
this can just generate 1 block, the 4 blocks is because i wanted to use same scenario for sync and send tests. we can update that now and create a 10block scenario with no sends for the sync test.